rbac/providers/azure/delta_outbound_sync.py

```diff
@@ -40,150 +44,92 @@ def getenv(name, default):
-            )
-            time.sleep(DELAY)
-        except Exception as err:
-            LOGGER.warning(err.__class__.__name__)
-            raise err
-
-
-def peek_at_queue():
-    """Gets the oldest item from the queue without deleting it."""
-    try:
-        queue_entry = (
-            r.table(OUTBOUND_QUEUE)
-            .filter({"provider_id": "azure"})
-            .min("timestamp")
-            .run()
-        )
-        return queue_entry
-    except (r.ReqlNonExistenceError, r.ReqlOpFailedError, r.ReqlDriverError) as err:
-        raise ExpectedError(err)
-    except Exception as err:
-        LOGGER.warning(type(err).__name__)
-        raise err
-
-
-def check_entry_AAD(queue_entry):
-    """Routes the given query entry to the proper handler to check if it already
+def is_entry_in_aad(queue_entry):
+    """Routes the given queue entry to the proper handler to check if it already
     exists in Azure AD."""
     data_type = queue_entry["data_type"]
-    if data_type == "USER":
-        check_user_AAD(queue_entry)
-    elif data_type == "GROUP":
-        check_group_AAD(queue_entry)
+    if data_type == "user":
+        is_user_in_aad(queue_entry)
+    elif data_type == "group":
+        is_group_in_aad(queue_entry)
 
 
-def check_group_AAD(queue_entry):
-    """Takes in a queue entry containing a group object and checks if it exists
-    in azure AD."""
-    group = queue_entry["data"]
-    response = fetch_group_AAD(group["id"])
+def is_user_in_aad(queue_entry):
+    """Takes in a queue entry containing a user object and checks if it exists
+    in azure AD. Returns True if the user exists, False if the user doesn't exist.
+    """
+    user = queue_entry["data"]
+    if user["user_id"]:
+        user_id = user["user_id"]
+    else:
+        user_id = user["user_principal_name"]
+    response = fetch_user_aad(user_id)
     if response["status_code"] >= 200 and response["status_code"] < 300:
         return True
-    elif response.status_code == 404:
+    elif response["status_code"] == 404:
         return False
     else:
         raise Exception(
             f"Error getting user in Azure AD: Status code {response.status_code}"
         )
 
 
-def check_user_AAD(queue_entry):
-    """Takes in a queue entry containing a user object and checks if it exists
-    in azure AD."""
-    user = queue_entry["data"]
-    if user["user_id"]:
-        user_id = user["user_id"]
-    else:
-        user_id = user["user_principal_name"]
-    response = fetch_user_AAD(user_id)
+def is_group_in_aad(queue_entry):
+    """Takes in a queue entry containing a group object and checks if it exists
+    in azure AD. Returns True if the group exists, False if the user doesn't exist.
+    """
+    group = queue_entry["data"]
+    response = fetch_group_aad(group["id"])
     if response["status_code"] >= 200 and response["status_code"] < 300:
         return True
-    elif response["status_code"] == 404:
+    elif response.status_code == 404:
```

Comment:
Why the dot separator here instead of key lookup like the other places?

Reply:
This changed happened on accident, I'll pushing out a change to revert it back.

Reply:
Change has been pushed.